discord_slash/client.py

```diff
@@ -410,9 +411,30 @@ async def sync_all_commands(
                 self.logger.debug(
                     f"Detected changes on {scope if scope is not None else 'global'}, updating them"
                 )
-                existing_cmds = await self.req.put_slash_commands(
-                    slash_commands=to_send, guild_id=scope
-                )
+                try:
+                    existing_cmds = await self.req.put_slash_commands(
+                        slash_commands=to_send, guild_id=scope
+                    )
+                except discord.HTTPException as ex:
+                    if ex.status == 400:
+                        # catch bad requests
+                        cmd_nums = set(
+                            re.findall(r"In\s(\d).", ex.args[0])
```

Comment:
What do you think about using `re.compile` it beforehand?

Reply:
Due to how rarely this code should execute I didn't feel it was necessary.

No harm in adding it, but also very little benefit

Reply:
true, its your call lol

Reply:
I'll add it. I guess theoretically someone could have hundreds of commands and that could take a while, relatively, to process

Reply:
This code executes once per sync, on error. In which case it stops execution and raises an error. It only runs the regex once. So I won't add compile to it

(putting this here to explain my logic if it comes up later)

Reply:
Yea I get what you mean, the regex is only used if their program is bound to not work anyways.